src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -409,6 +411,10 @@ public function has($id)
      */
     public function get($id, $invalidBehavior = ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE)
     {
+        if (!$this->compiled) {
+            @trigger_error(sprintf('Calling %s before compiling the container is deprecated since version 3.2 and will throw an exception in 4.0.', __METHOD__), E_USER_DEPRECATED);
+        }
```

Comment:
We may want to think about moving this down after the call to the parent class' `get()` method (so you would be able to retrieve services that have been added using `set()` before). Alternatively, deprecating `set()` too might be another way to go.